dandiapi/api/views/dandiset.py

```diff
@@ -69,51 +72,75 @@ class DandisetFilterBackend(filters.OrderingFilter):
-                    '-created'
-                )[:1]
-                # get the `modified` field of the most recent version.
-                # '_version' is appended because the Dandiset model already has a `modified` field
-                queryset = queryset.annotate(
-                    modified_version=Subquery(latest_version.values('modified'))
-                )
-                return queryset.order_by(f'{ordering}_version')
-            if ordering.endswith('size'):
-                latest_version = Version.objects.filter(dandiset=OuterRef('pk')).order_by(
-                    '-created'
-                )[:1]
-                queryset = queryset.annotate(
-                    size=Subquery(
-                        latest_version.annotate(
-                            size=Coalesce(Sum('assets__blob__size'), 0)
-                            + Coalesce(Sum('assets__zarr__size'), 0)
-                        ).values('size')
-                    )
+        if not orderings:
+            return queryset
+        ordering = orderings[0]
+
+        # ordering can be either 'created' or '-created', so test for both
+        if ordering.endswith('id'):
+            return queryset.order_by(ordering)
+
+        if ordering.endswith('name'):
+            # name refers to the name of the most recent version, so a subquery is required
+            latest_version = Version.objects.filter(dandiset=OuterRef('pk')).order_by('-created')[
+                :1
+            ]
+            queryset = queryset.annotate(name=Subquery(latest_version.values('metadata__name')))
+            return queryset.order_by(ordering)
+
+        if ordering.endswith('modified'):
+            # modified refers to the modification timestamp of the most
+            # recent version, so a subquery is required
+            latest_version = Version.objects.filter(dandiset=OuterRef('pk')).order_by('-created')[
+                :1
+            ]
+            # get the `modified` field of the most recent version.
+            # '_version' is appended because the Dandiset model already has a `modified` field
+            queryset = queryset.annotate(
+                modified_version=Subquery(latest_version.values('modified'))
+            )
+            return queryset.order_by(f'{ordering}_version')
+
+        if ordering.endswith('size'):
+            latest_version = Version.objects.filter(dandiset=OuterRef('pk')).order_by('-created')[
+                :1
+            ]
+            queryset = queryset.annotate(
+                size=Subquery(
+                    latest_version.annotate(
+                        size=Coalesce(Sum('assets__blob__size'), 0)
+                        + Coalesce(Sum('assets__zarr__size'), 0)
+                    ).values('size')
                 )
-                return queryset.order_by(ordering)
+            )
+            return queryset.order_by(ordering)
+
```

Comment:
Why this big change? This refactors the function to have an early exit (and therefore better structure) without changing its semantics, right?

Reply:
Correct. I was already touching this part of the code (changing `DandisetFilterBackend` to `DandisetOrderingFilter`) and took it upon myself to make this change for code readability.

Reply:
Ok. I'd have preferred the unrelated change go in its own commit (or even its own PR) but I won't insist on that now that the thing is done.